### ISSUE TITLE
feat: add stochastic (online/mini-batch) mode to LogisticRegression.fit()

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#428]
## Problem

`LogisticRegression::fit()` uses full-batch gradient descent (computes gradient over all samples, then averages). For imbalanced binary classification with class weighting, this dilutes minority class signal because:

1. Gradient is averaged over n_samples
2. With 92% safe / 8% unsafe, even with sqrt-inverse class weights, ~90% of the averaged gradient comes from safe examples
3. Result: model converges to "always predict majority class"

Empirically: MCC drops from 0.47 (1000 entries) to 0.17 (4000 entries) despite class weighting.

## Solution

Add a `FitMode` enum to control optimization strategy:

```rust
pub enum FitMode {
    /// Full-batch gradient descent (current behavior, default)
    Batch,
    /// Online SGD: update weights after each sample
    Stochastic,
    /// Mini-batch: update per mini-batch of given size
    MiniBatch(usize),
}

let mut model = LogisticRegression::new()
    .with_fit_mode(FitMode::Stochastic)
    .with_class_weight(ClassWeight::Balanced);
```

### Why online SGD helps

Per-sample updates are more responsive to class weights:
- Each minority sample immediately adjusts weights (no dilution by majority averaging)
- This is exactly how scikit-learn's SGDClassifier works
- Online SGD with class weights at n=1500: MCC=0.324 PASS (vs batch GD: 0.222 FAIL)

## Upstream consumer

bashrs SSC classifier pipeline. Currently uses hand-rolled online SGD because batch GD fails at scale.

Depends on: #427 (ClassWeight + L2, now merged)
